modules/aardvarkBidAdapter.js

```diff
@@ -0,0 +1,154 @@
+    }
+
+    utils._each(validBidRequests, function(b) {
+      var rMap = requestsMap[b.params.ai];
+      if (!rMap) {
+        rMap = {
+          shortCodes: [],
+          payload: {
+            version: 1,
+            jsonp: false,
+            rtkreferer: referer
+          },
+          endpoint: DEFAULT_ENDPOINT
+        };
+
+        if (pageCategories && pageCategories.length) {
+          rMap.payload.categories = pageCategories.slice(0);
+        }
+
+        if (b.params.categories && b.params.categories.length) {
+          rMap.payload.categories = rMap.payload.categories || []
+          utils._each(b.params.categories, function(cat) {
+            rMap.payload.categories.push(cat);
+          });
+        }
+
+        if (bidderRequest && bidderRequest.gdprConsent) {
+          rMap.payload.gdpr = bidderRequest.gdprConsent.gdprApplies;
+          rMap.payload.consent = bidderRequest.gdprConsent.consentString;
+        }
+
+        requestsMap[b.params.ai] = rMap;
+        auctionCodes.push(b.params.ai);
+      }
+
+      rMap.shortCodes.push(b.params.sc);
+      rMap.payload[b.params.sc] = b.bidId;
+
+      if ((typeof b.params.host === 'string') && b.params.host.length &&
+          (b.params.host !== rMap.endpoint)) {
+        rMap.endpoint = b.params.host;
+      }
+    });
+
+    utils._each(auctionCodes, function(auctionId) {
+      var req = requestsMap[auctionId];
+      requests.push({
+        method: 'GET',
+        url: `//${req.endpoint}/${auctionId}/${req.shortCodes.join('_')}/aardvark`,
+        data: req.payload,
+        bidderRequest
+      });
+    });
+
+    return requests;
+  },
+
+  interpretResponse: function(serverResponse, bidRequest) {
+    var bidResponses = [];
+    utils._each(serverResponse.body, function(rawBid) {
+      if (!rawBid.cpm) {
+        return;
+      }
+
+      var bidResponse = {
+        requestId: rawBid.cid,
+        cpm: rawBid.cpm,
+        width: rawBid.width || 0,
+        height: rawBid.height || 0,
+        currency: rawBid.currency ? rawBid.currency : AARDVARK_CURRENCY,
+        netRevenue: rawBid.netRevenue ? rawBid.netRevenue : true,
+        ttl: rawBid.ttl ? rawBid.ttl : AARDVARK_TTL,
+        exchange: rawBid.ex,
+        creativeId: rawBid.creativeId || 0
+      };
+
+      if (rawBid.hasOwnProperty('dealId')) {
+        bidResponse.dealId = rawBid.dealId
+      }
+
+      switch (rawBid.media) {
+        case 'banner':
+          bidResponse.ad = rawBid.adm + utils.createTrackPixelHtml(decodeURIComponent(rawBid.nurl));
+          break;
+
+        default:
+          return utils.logError('bad Aardvark response (media)', rawBid);
+      }
+
+      bidResponses.push(bidResponse);
+    });
+
+    return bidResponses;
+  },
+
+  getUserSyncs: function(syncOptions, serverResponses, gdprConsent) {
+    const syncs = [];
+    var url = '//' + SYNC_ENDPOINT + '/cs';
+    var gdprApplies = true;
+    if (typeof gdprConsent.gdprApplies === 'boolean') {
```

Comment:
please add to the check if `gdprConsent` exists.  On pages without a CMP, this current code would lead to an undefined error.